src/indicators/drought/arc2_precipitation.py

```diff
@@ -0,0 +1,710 @@
+
+        with open(raw_filepath, "wb") as fd:
+            for chunk in response.iter_content(chunk_size=128):
+                fd.write(chunk)
+
+        # merge to master and delete temporary files
+        if not master:
+            master_filepath = self._get_raw_filepath(master=True)
+            if not os.path.isfile(master_filepath):
+                logger.error(
+                    (
+                        "Master raster data file not set. This needs "
+                        "to be done once by running "
+                        "`download_data(master=True)` "
+                        "for a unique set of coordinates."
+                    )
+                )
+
+            logger.info(
+                f"Merging ARC2 data from {self.date_min:%d %b %Y} "
+                f"to {self.date_max:%d %b %Y} "
+                f"into master file: {os.path.basename(master_filepath)}."
+            )
+
+            with xr.open_dataarray(master_filepath) as ds:
+                master = ds.load()
+            with xr.open_dataarray(raw_filepath) as ds:
+                raw = ds.load()
+
+            master_merge = xr.concat([master, raw], dim="T")
+
+            os.remove(raw_filepath)
+            os.remove(master_filepath)
+
+            if "_FillValue" in master_merge.encoding and np.isnan(
+                master_merge.encoding["_FillValue"]
+            ):
+                master_merge.encoding["_FillValue"] = -999
+
+            master_merge.to_netcdf(master_filepath)
+
+        return
+
+    def _get_directory(self, dir: Union[Path, str]) -> Path:
+        """
+        Return ARC2 directory, either in public, processed
+        or raw for a specific ISO3.
+
+        :param dir: Path or string of folder name in public for passing.
+        """
+        directory = (
+            DATA_DIR / PUBLIC_DATA_DIR / dir / self.country_iso3 / ARC2_DIR
+        )
+        return directory
+
+    def _get_raw_filepath(self, master: bool) -> Path:
+        """
+        Return filepath to raw ARC2 data for specific x
+        and y bounds. All data stored within a single
+        master file. If `master`, the filepath is not
+        specified for a specific date range and designated
+        with 'master.nc' at the end. Otherwise, end of
+        filepath designated with `self.date_min` and
+        `self.date_max`.
+
+        :param master: Whether or not the filepath is
+            for the master file.
+        :param temp: Whether or not the filepath is
+            for a temp file used for saving.
+        """
+        directory = self._get_directory(RAW_DATA_DIR)
+
+        if master:
+            end = "master"
+        else:
+            end = f"{self.date_min:%d_%b_%Y}_{self.date_max:%d_%b_%Y}"
+
+        filename = (
+            f"arc2_daily_precip_{self.country_iso3}_"
+            f"{self.range_x[0]}_{self.range_x[1]}_"
+            f"{self.range_y[0]}_{self.range_y[1]}_{end}.nc"
+        )
+        return directory / Path(filename)
```

Comment:
Once the first part i.e. `directory` is a path, you don't have to convert the string into a Path before concatenating